### DISCUSSION TITLE
How complicated is it to switch from UWP to Avalonia ?

### DISCUSSION THREAD

**Discussion by Zilounay** [AvaloniaUI/Avalonia#5804] [Q&A]
Hello,

I couldn't find a discussion about this online. I have a fairly simple project in terms of UI, currently developed in UWP and mostly consisting of filtered/ordered tables, tab-navigation, and some other elements like buttons & labels.

I'm fairly confident avalonia can handle all that easily, but the question I have is : will I get into trouble if I want to switch from UWP to avalonia ? I'm in the process of investigating if I'm better off rewriting from scratch, that process starts by simply asking here. I'm assuming my best bet is to create an avalonia project and move most of the code there until I reach the UI level ; but if there is any key information to know beforehand, I would love to hear it !

Hopefully you guys have some insight.

Thank you all for your time

**Comment by danwalmsley** [comment]
If you have followed MVVM and used Viewmodels then it should be quite easy. If alot if your code is code behind it will be harder.

Although Avalonia is closer to WPF than UWP we have a fair few UWPike apis I.e. FlyOuts and ItemsRepeater.

So not impossible, often quite easy. Depends on how you have architected your original app.
